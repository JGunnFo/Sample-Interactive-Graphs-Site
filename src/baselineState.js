
export const initialStateReducer={
    section: "Main",
    transition:false
  };

export const initialStateTest={
    section: "Main",
    transition:false
};


export const initialStateTestAbout={
  section: "About",
  transition:false
};