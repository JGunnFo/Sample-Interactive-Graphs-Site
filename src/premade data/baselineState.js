import {Articles, HTMLs, Graphs, Forms} from './articles';

export const initialStateReducer={
    section: "Article",
    transition:false,
    articleList:Articles,
    currentArticle:["HTML3", "GraphArranged_Genres_August", "HTML2", "GraphDate_Week2_August", "GraphArranged_Genres_August"],
    /*
    ^ie the HTMLGraphList
    v they use keys to get each
    */
    graphList: Graphs,
    formList: Forms,
    HTMLList: HTMLs,
    currentGraphs: [ ["GraphMovie_Byrnadel_August","GraphArranged_Genres_August"], ["GraphArranged_Genres_August","GraphDate_Week3_August"], ["GraphMovie_Byrnadel_August","GraphArranged_Genres_August"]]
    /* current, origin. 0 is current, 1 is origin*/

  };

  /*
  so currentGraphs doesnt store the graphs, 
  it stores a pair of string references like Current:Origin 
  so ["Graph2":"Graph1"], and then so graph function refers to the original, 
  but functions will reset the current to the original
  */

  

export const initialStateTest={
  section: "News",
  transition:false,
  articleList:Articles,
  currentArticle:["HTML3", "GraphArranged_Genres_August", "HTML2", "GraphDate_Week2_August", "GraphArranged_Genres_August"],
  /*
  ^ie the HTMLGraphList
  v they use keys to get each
  */
  graphList: Graphs,
  formList: Forms,
  HTMLList: HTMLs,
  currentGraphs: [ ["GraphMovie_Byrnadel_August","GraphArranged_Genres_August"], ["GraphArranged_Genres_August","GraphDate_Week3_August"], ["GraphMovie_Byrnadel_August","GraphArranged_Genres_August"]]
  /* current, origin. 0 is current, 1 is origin*/
};


export const initialStateTestAbout={
  section: "About",
  transition:false
};