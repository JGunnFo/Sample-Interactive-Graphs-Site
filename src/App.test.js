import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from "react-redux";
import { render } from '@testing-library/react';
import TopComponent from './App';
import {initialStateTestNews, initialStateTestAbout, initialStateTestArticleGraphSet1, initialStateTestArticleGraphSet2} from './premade data/baselineState';
import ReactTestUtils from 'react-dom/test-utils'; // ES6



/*
The "click on any of the bars below..." line will not show up in the snapshots before graphs
unless HTML1 is included as well, so don't be surprised by that.
Maybe that should be moved to being part of the ChartMaker function, 
but for now it is unimportant and longterm it is ambiguous.
*/

function genericSnapshotTestFunction(name, initialState){


describe(name, () => {

  let App
  let mockStore
  let theStore

  beforeAll(() => {
    
  mockStore = configureMockStore()
  theStore=mockStore(initialState)

  App = () => (
    <Provider store={theStore}>
        <TopComponent />
    </Provider>
)
  });
  

  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(2);
  });


  
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
it('renders correctly according to snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});




});

}

genericSnapshotTestFunction('<App starting with News>', initialStateTestNews)
genericSnapshotTestFunction('<App starting with About>', initialStateTestAbout)
genericSnapshotTestFunction('<App starting with Article Graph Set 1>', initialStateTestArticleGraphSet1)
genericSnapshotTestFunction('<App starting with Article Graph Set 2>', initialStateTestArticleGraphSet2)
