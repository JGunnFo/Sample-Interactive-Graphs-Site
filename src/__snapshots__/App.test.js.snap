// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App starting with About> renders correctly according to snapshot 1`] = `
<div
  className="Overall-Parent"
>
  <div>
    <div>
      <div>
        <div
          className="Nav-Flex-Parent"
          role="navigation"
        >
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  News
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  Article
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className="Nav-Selected"
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  About
                </button>
              </div>
            </div>
          </div>
          <div>
            <div>
              <a
                aria-label="Github Link"
                className="Nav-Link"
                href="https://github.com/JGunnFo?tab=repositories"
              >
                Github
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
  >
    <div>
      <div>
        <div
          role="main"
        >
          <div
            className="Article-HTML-Segment"
          >
            <div>
              This is a demonstratory fake business website made by Jeffrey Gunn, a React Redux web developer.
            </div>
            <div>
              I'll be adding more to this site over time. More details on the site are on the 
              <a
                aria-label="Github Link"
                href="https://github.com/JGunnFo/sampleBiz"
              >
                github page.
              </a>
            </div>
            <div>
              I'll also be updating my github and my 
              <a
                aria-label="Home Site Link"
                href="https://jeffrey-gunn.com"
              >
                home page
              </a>
               with new projects and revised projects.
            </div>
            <div>
              <a
                aria-label="Mail Jeffrey Gunn"
                href="mailto:jeffgunnweb@gmail.com?subject=Edit this subject line to avoid the filter for bots."
              >
                Contact me
              </a>
               if you're looking for web developers, especially React and/or Redux ones.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App starting with Article Graph Set 1> renders correctly according to snapshot 1`] = `
<div
  className="Overall-Parent"
>
  <div>
    <div>
      <div>
        <div
          className="Nav-Flex-Parent"
          role="navigation"
        >
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  News
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className="Nav-Selected"
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  Article
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  About
                </button>
              </div>
            </div>
          </div>
          <div>
            <div>
              <a
                aria-label="Github Link"
                className="Nav-Link"
                href="https://github.com/JGunnFo?tab=repositories"
              >
                Github
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
  >
    <div>
      <div>
        <div
          role="main"
        >
          <div
            className="Article-Parent"
          >
            <div>
              <div>
                <div
                  className="Article-HTML-Segment"
                >
                  <div>
                    <div>
                      In October we will be holding preview screenings for 6 different upcoming movies! 
                    </div>
                    <div>
                      Fill out the form below with the absolutely not fake email that you use for your Fictional Movie Business account, your favorite movie genre, and your favorite movie snack, and we'll get back to you soon!
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="Screenreader-Only"
                  >
                     Here is a readout of this page's graph:
                  </div>
                  <div
                    role="presentation"
                  >
                    <div
                      aria-live="polite"
                    >
                      <div
                        className="Graph-Title"
                      >
                        August Sales By Genre
                      </div>
                      <div
                        className="Screenreader-Only"
                      >
                        <div>
                          <div>
                            <div>
                              <div>
                                Horror
                                  
                                sales
                                : 
                                400
                              </div>
                            </div>
                            <div>
                              <div>
                                Action
                                  
                                sales
                                : 
                                700
                              </div>
                            </div>
                            <div>
                              <div>
                                Romance
                                  
                                sales
                                : 
                                600
                              </div>
                            </div>
                            <div>
                              <div>
                                Comedy
                                  
                                sales
                                : 
                                650
                              </div>
                            </div>
                            <div>
                              <div>
                                Drama
                                  
                                sales
                                : 
                                550
                              </div>
                            </div>
                          </div>
                          You can receive a new set of data based on each of these. Navigate the site with your keyboard until you arrive on data you want to learn more about, then hit enter. That will generate a new set of data which will be read. For example, if the Graph is "August Movie Sales by Genre" and then you select "Action sales", you will receive data about individual action movies' sales in August.
                        </div>
                      </div>
                    </div>
                    <div
                      aria-label="Graph"
                      className="Graph-Overall"
                    >
                      <div
                        className="recharts-responsive-container"
                        id="undefined"
                        style={
                          Object {
                            "height": 400,
                            "maxHeight": undefined,
                            "minHeight": undefined,
                            "minWidth": undefined,
                            "width": "99%",
                          }
                        }
                      >
                        <div />
                      </div>
                    </div>
                  </div>
                  <div
                    className="Revert-Graph-Parent"
                  >
                    <button
                      className="Revert-Graph"
                      onClick={[Function]}
                    >
                      Revert Graph
                    </button>
                  </div>
                </div>
                <div
                  className="Article-HTML-Segment"
                >
                  <div>
                    If you're sad to see your favorite genre did not do as well as you'd like, you know what to do: Give us more money!
                  </div>
                </div>
                <div>
                  <div
                    className="Screenreader-Only"
                  >
                     Here is a readout of this page's graph:
                  </div>
                  <div
                    role="presentation"
                  >
                    <div
                      aria-live="polite"
                    >
                      <div
                        className="Graph-Title"
                      >
                        Week 2 of August's Movie Sales by Genre
                      </div>
                      <div
                        className="Screenreader-Only"
                      >
                        <div>
                          <div>
                            <div>
                              <div>
                                Horror
                                  
                                sales
                                : 
                                70
                              </div>
                            </div>
                            <div>
                              <div>
                                Action
                                  
                                sales
                                : 
                                150
                              </div>
                            </div>
                            <div>
                              <div>
                                Romance
                                  
                                sales
                                : 
                                250
                              </div>
                            </div>
                            <div>
                              <div>
                                Comedy
                                  
                                sales
                                : 
                                300
                              </div>
                            </div>
                            <div>
                              <div>
                                Drama
                                  
                                sales
                                : 
                                100
                              </div>
                            </div>
                          </div>
                          You can receive a new set of data based on each of these. Navigate the site with your keyboard until you arrive on data you want to learn more about, then hit enter. That will generate a new set of data which will be read. For example, if the Graph is "August Movie Sales by Genre" and then you select "Action sales", you will receive data about individual action movies' sales in August.
                        </div>
                      </div>
                    </div>
                    <div
                      aria-label="Graph"
                      className="Graph-Overall"
                    >
                      <div
                        className="recharts-responsive-container"
                        id="undefined"
                        style={
                          Object {
                            "height": 400,
                            "maxHeight": undefined,
                            "minHeight": undefined,
                            "minWidth": undefined,
                            "width": "99%",
                          }
                        }
                      >
                        <div />
                      </div>
                    </div>
                  </div>
                  <div
                    className="Revert-Graph-Parent"
                  >
                    <button
                      className="Revert-Graph"
                      onClick={[Function]}
                    >
                      Revert Graph
                    </button>
                  </div>
                </div>
                <div>
                  <div
                    className="Screenreader-Only"
                  >
                     Here is a readout of this page's graph:
                  </div>
                  <div
                    role="presentation"
                  >
                    <div
                      aria-live="polite"
                    >
                      <div
                        className="Graph-Title"
                      >
                        Byrnadel's Squadron's Sales in August
                      </div>
                      <div
                        className="Screenreader-Only"
                      >
                        <div>
                          <div>
                            <div>
                              <div>
                                Week 1
                                  
                                sales
                                : 
                                100
                              </div>
                            </div>
                            <div>
                              <div>
                                Week 2
                                  
                                sales
                                : 
                                70
                              </div>
                            </div>
                            <div>
                              <div>
                                Week 3
                                  
                                sales
                                : 
                                50
                              </div>
                            </div>
                            <div>
                              <div>
                                Week 4
                                  
                                sales
                                : 
                                30
                              </div>
                            </div>
                          </div>
                          You can receive a new set of data based on each of these. Navigate the site with your keyboard until you arrive on data you want to learn more about, then hit enter. That will generate a new set of data which will be read. For example, if the Graph is "August Movie Sales by Genre" and then you select "Action sales", you will receive data about individual action movies' sales in August.
                        </div>
                      </div>
                    </div>
                    <div
                      aria-label="Graph"
                      className="Graph-Overall"
                    >
                      <div
                        className="recharts-responsive-container"
                        id="undefined"
                        style={
                          Object {
                            "height": 400,
                            "maxHeight": undefined,
                            "minHeight": undefined,
                            "minWidth": undefined,
                            "width": "99%",
                          }
                        }
                      >
                        <div />
                      </div>
                    </div>
                  </div>
                  <div
                    className="Revert-Graph-Parent"
                  >
                    <button
                      className="Revert-Graph"
                      onClick={[Function]}
                    >
                      Revert Graph
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App starting with Article Graph Set 2> renders correctly according to snapshot 1`] = `
<div
  className="Overall-Parent"
>
  <div>
    <div>
      <div>
        <div
          className="Nav-Flex-Parent"
          role="navigation"
        >
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  News
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className="Nav-Selected"
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  Article
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  About
                </button>
              </div>
            </div>
          </div>
          <div>
            <div>
              <a
                aria-label="Github Link"
                className="Nav-Link"
                href="https://github.com/JGunnFo?tab=repositories"
              >
                Github
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
  >
    <div>
      <div>
        <div
          role="main"
        >
          <div
            className="Article-Parent"
          >
            <div>
              <div>
                <div
                  className="Article-HTML-Segment"
                >
                  <div>
                    <div>
                      On Saturday, September 5th, all tickets will be 30% off! That applies to both booking in advance, day-of purchases, and digital screenings! The only thing you have to do is show up and buy at least 2 buckets of popcorn.
                    </div>
                  </div>
                </div>
                <div
                  className="Article-HTML-Segment"
                >
                  <div>
                    <div>
                      Here at Fictional Movie Business, we've gone to great extents to allow people to bring all of their pets to our theaters, including immense effort to install animal-only areas and to provide trained staff to make sure nothing goes wrong. We love to see all your dogs, cats, lizards, alpacas, ferrets, and everything else!
                    </div>
                    <div>
                      <br />
                      However, for you to have a safe and pleasant moviegoing experience, we have added 3 new rules:
                    </div>
                    <div>
                      -All animals must be at least a year old, so that they have any hope of obeying rules.
                    </div>
                    <div>
                      -If the movie is 3 hours or longer, you must bring snacks for our staff to feed to your pet.
                    </div>
                    <div>
                      -No leash, no service.
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="Screenreader-Only"
                  >
                     Here is a readout of this page's graph:
                  </div>
                  <div
                    role="presentation"
                  >
                    <div
                      aria-live="polite"
                    >
                      <div
                        className="Graph-Title"
                      >
                        Outcasts In Love's Sales in August
                      </div>
                      <div
                        className="Screenreader-Only"
                      >
                        <div>
                          <div>
                            <div>
                              <div>
                                Week 1
                                  
                                sales
                                : 
                                20
                              </div>
                            </div>
                            <div>
                              <div>
                                Week 2
                                  
                                sales
                                : 
                                20
                              </div>
                            </div>
                            <div>
                              <div>
                                Week 3
                                  
                                sales
                                : 
                                5
                              </div>
                            </div>
                            <div>
                              <div>
                                Week 4
                                  
                                sales
                                : 
                                25
                              </div>
                            </div>
                          </div>
                          You can receive a new set of data based on each of these. Navigate the site with your keyboard until you arrive on data you want to learn more about, then hit enter. That will generate a new set of data which will be read. For example, if the Graph is "August Movie Sales by Genre" and then you select "Action sales", you will receive data about individual action movies' sales in August.
                        </div>
                      </div>
                    </div>
                    <div
                      aria-label="Graph"
                      className="Graph-Overall"
                    >
                      <div
                        className="recharts-responsive-container"
                        id="undefined"
                        style={
                          Object {
                            "height": 400,
                            "maxHeight": undefined,
                            "minHeight": undefined,
                            "minWidth": undefined,
                            "width": "99%",
                          }
                        }
                      >
                        <div />
                      </div>
                    </div>
                  </div>
                  <div
                    className="Revert-Graph-Parent"
                  >
                    <button
                      className="Revert-Graph"
                      onClick={[Function]}
                    >
                      Revert Graph
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App starting with News> renders correctly according to snapshot 1`] = `
<div
  className="Overall-Parent"
>
  <div>
    <div>
      <div>
        <div
          className="Nav-Flex-Parent"
          role="navigation"
        >
          <div>
            <div
              className="Nav-Selected"
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  News
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  Article
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              <div>
                <button
                  className="Button-Unset"
                  onClick={[Function]}
                >
                  About
                </button>
              </div>
            </div>
          </div>
          <div>
            <div>
              <a
                aria-label="Github Link"
                className="Nav-Link"
                href="https://github.com/JGunnFo?tab=repositories"
              >
                Github
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
  >
    <div>
      <div>
        <div
          role="main"
        >
          <div>
            <div>
              <div
                className="News-Grid-Parent"
              >
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        Interactive Data
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        Ever wonder if horror movies are more popular than romance movies?  These August movie sales will tell the truth!
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        Sign up for a special screening!
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        If you fill out the form, you'll get to attend a secret preview screening of an upcoming movie!
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        Updated Rules on Bringing Pets
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        If you want to bring your alpaca along, be sure to read these rules to save your movie night!
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        30% off Tickets on September 5th!
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        What could be better than cheap movies on the weekend?
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        Several 1970s movies available!
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        We've added a lot of 1970s classics to our digital library!
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        Horror Week Sale!
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        For horror week, you can save big! Click here to find out more!
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        T-Shirts with your favorite brand!
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        You love us, right? Then wear a t-shirt with our logo on it! You'll get a ticket discount!
                      </div>
                    </div>
                  </button>
                </div>
                <div>
                  <button
                    className="Button-Unset News-Square-Button"
                    onClick={[Function]}
                  >
                    <div
                      className="News-Square-Div"
                    >
                      <div
                        className="News-Title-Text"
                      >
                        Staff Changes
                      </div>
                      <div
                        className="News-Preview-Text"
                      >
                        We've got some new hires, and some staff relocating.
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
